src/mtstate.c

```diff
@@ -33,19 +33,32 @@ static int inline touch_range_ratio(const struct MConfig* cfg, int value)
 	return (double)(value - cfg->touch_min) / (double)(cfg->touch_max - cfg->touch_min) * 100;
 }
 
+static int inline pressure_range_ratio(const struct MConfig* cfg, int value)
+{
+	return percentage(value - cfg->pressure_min, cfg->pressure_max - cfg->pressure_min);
+}
+
+static int finger_touch_ratio(const struct MConfig* cfg, const struct FingerState* hw)
+{
+	switch(cfg->touch_type){
+	case MCFG_SCALE:
+		return percentage(hw->touch_major, hw->width_major); /* = estimated pressure */
+	case MCFG_SIZE:
+	case MCFG_SIZE_PRESSURE:
+		return touch_range_ratio(cfg, hw->touch_major);
+	case MCFG_PRESSURE_SIZE:
+	case MCFG_PRESSURE:
+		return pressure_range_ratio(cfg, hw->pressure);
+	default: return 101; /* sholuld it be additional argument? or maybe it should return -1? */
```

Comment:
"default" label is unreachable, otherwise touch_type would have invalid value.

Returned value represents percentage, normally it would have range of <0, 100>. I chose value that is out of range, and produces desired result when compared with touch_down and touch_up.